src/pages/Detail/DetailTop.js

```diff
@@ -1,54 +1,121 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 
+import AmountOption from './AmountOption';
+
+// select -> state
+// 함수를 만들어서 자식한테 전달
+// 자식에서 함수기 실행되면은 select된 값을 변경
+
 export class DetailTop extends Component {
+  //state초기화
+  constructor(props) {
+    super(props);
+    this.state = {
+      numValue: 1,
+      isOn: false,
+      amountOption: [],
+      selectedOption: this.props.productOption[0],
+      productOption: '',
+      option: '',
+    };
+    this.onClickToggleHandler = this.onClickToggleHandler.bind(this);
+  }
+  //수량 증가 감소
+  onClickHandler = () => {
+    this.setState({ numValue: this.state.numValue + 1 });
+  };
+  onClickhandlerRemove = () => {
+    //const { numValue } = this.state.numValue;
+    if (this.state.numValue > 1) {
+      this.setState({ numValue: this.state.numValue - 1 });
+    }
+  };
+  //종류 추가
+  onClickToggleHandler = () => {
+    //const { isOn } = this.state.isOn;
+    //console.log(isOn);
+    this.setState({
+      isOn: !this.state.isOn,
+    });
+  };
+
+  selectOption = option => {
+    this.setState({
+      selectedOption: option,
+    });
+    this.onClickToggleHandler();
+  };
+
   render() {
+    const productInfo = this.props;
+    const selectedOption = this.state.selectedOption;
```

Comment:
```suggestion
    const { selectedOption } = this.state;
```

Reply:
필요 없는 부분은 삭제하였습니다!감사합니다
```suggestion
    const selectedOption = this.state.selectedOption;
```

Reply:
- 비구조화 할당 문법은 다음과 같습니다!
```suggestion
    const { selectedOption } = this.state;
```